### PR TITLE
DM-30266 Adapt to UUID use for QuantumGraph nodeIds

### PR DESCRIPTION
**Pull request by natelust**

### CODE REVIEW

File: bin/run_demo.sh [hunk 1]
```diff
@@ -109,18 +109,21 @@ refresh_exe
 # Run the init step
 pipetask --long-log run -b "$tmp_butler" -i "$incoll" --output-run "$exerun" --init-only --register-dataset-types --qgraph "$graph_file" --extend-run
 
+NODES=( $(pipetask qgraph -b "$tmp_butler" -g "$graph_file" --show-qgraph-header \
+    |python -c 'import json,sys;nodes=[id for id, _ in json.load(sys.stdin)["Nodes"]];print(" ".join(nodes))') )
```
Comment (timj) [inline bin/run_demo.sh:113]: Consider using `jq` here for extracting the relevant content from the JSON. It's optimized for that.

Comment (natelust) [reply bin/run_demo.sh:113]: That would have been my first choice, but jq is not on every system, so I went with something that was

Comment (timj) [reply bin/run_demo.sh:113]: it's part of rubin-env so should be fine but I'll defer to @ktlim in case he has a preference.

Comment (ktlim) [reply bin/run_demo.sh:113]: Definitely suitable for `pipelines_check`.  And it's a pretty minimal dependency (single binary, no runtime dependencies) for any non-Rubin use of G3MW.

**Review by TallJimbo — APPROVED** [review]
I don't have an opinion myself, but I'd defer to Tim and K-T on the `jq` question.

**Pull request merged**